src/Streamly/Prelude.hs

```diff
@@ -166,6 +168,16 @@ foldl step begin done m = go begin (toStream m)
             yield a (Just x) = go (step acc a) x
          in (runStream m1) Nothing stop yield
 
+-- | Scan left. A strict left fold which accumulates the result of its reduction steps inside a stream, from left.
+scanl :: (Streaming t, Monad m) => (b -> a -> b) -> b -> t m a -> t m b
```

Comment:
Let's make the signature on the lines of `foldl`, having a `done` function (`x -> b` in foldl) as well so that this works well with the `foldl` package. When writing a new prelude functions it may be a good idea to take a look at how the `streaming` package types it or implements it. `streaming` is pretty close to `streamly`.